highstreets/features/processing_functions.py

```diff
@@ -0,0 +1,154 @@
+import pandas as pd
+from sklearn.linear_model import LinearRegression
+
+
+# create function to take averages of every x months
+# and see which hs fall in which proportion x months after that
+def create_labels(hs, label_month, comparison_start, comparison_end):
+    """Labels high streets at risk, depending on whether
+    spending is in which quartile of average of x months prior"""
+    hs_latest = hs.loc[(hs["month_year"] == label_month)]
+    hs_latest = hs_latest.groupby(["highstreet_name"]).mean()
+    # turn comp start and comp end variables into dates and find sequence btween
+    comparison_start = pd.to_datetime(comparison_start)
+    comparison_end = pd.to_datetime(comparison_end)
+    date_range = pd.date_range(comparison_start, comparison_end, freq="m").to_period(
+        "M"
+    )
+    hs_compare = hs.loc[(hs["month_year"]).isin(date_range)]
+    hs_compare = hs_compare.groupby(["highstreet_name"]).mean()
+
+    # merge dataframe
+    merged_df = pd.merge(hs_latest, hs_compare, on="highstreet_name")
+    # condtional labelling, if latest value not at least at average level, label at risk
+    merged_df["proportion"] = (
+        merged_df["txn_amt_wd_retail_x"] / merged_df["txn_amt_wd_retail_y"]
+    )
+    merged_df["labels"] = pd.qcut(
+        merged_df["txn_amt_wd_retail_x"] / merged_df["txn_amt_wd_retail_y"],
+        4,
+        labels=[0, 1, 2, 3],
+    )
+    merged_df = merged_df.reset_index()
+    # subset relevant columns
+    merged_df = merged_df[["labels"]]
+    return merged_df
+
+
+# loop and select data for relevant months
+def create_mean_sd_o2(highstreet_df, predictor_days):
+    df = []
+    for days in predictor_days:
+        df.append(highstreet_df.loc[(highstreet_df["count_date"] == days)])
+    df = pd.concat(df)
+
+    # group to create mean and std across all months
+    mean = df.groupby(["highstreet_name"]).mean().reset_index()
+    # rename and merge so less possibility of mistakes
+    mean = mean.rename({"h13": "mean_o2"}, axis=1)
```

Comment:
Again maybe a good idea to just document assumptions about particular columns existing in the incoming data frame, h13 in this case, and highstreet_name (this isn't something I've done consistently, but probably a good habit for us to develop!)